setup.py

```diff
@@ -33,18 +33,21 @@
         )
         sys.exit(1)
 
+# We check Python's version ourselves in case someone installed Sopel on an
+# old version of pip (<9.0.0), which doesn't know about `python_requires`.
 if sys.version_info < (2, 7) or (
     sys.version_info.major >= 3 and sys.version_info < (3, 3)
 ):
     # Maybe not the best way to do this, but this question is tiring.
     raise ImportError('Sopel requires Python 2.7+ or 3.3+.')
 # Py2 EOL: https://www.python.org/dev/peps/pep-0373/#maintenance-releases
 if sys.version_info.major == 2:
-    if time.time() >= 1577836800:  # 2020-01-01 00:00:00 UTC
-        state = 'is near end of life'
+    now = time.time()
+    if now >= 1577836800:  # 2020-01-01 00:00:00 UTC
```

Comment:
Not introduced by this PR, but I think this logic is inverted, though since we're past the date already this conditional could be made static.